src/ServiceControl.Config/Commands/ForceUpgradeAuditInstanceCommand.cs

```diff
@@ -0,0 +1,140 @@
+namespace ServiceControl.Config.Commands;
+
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Caliburn.Micro;
+using Framework.Commands;
+using Events;
+using Framework.Modules;
+using Framework;
+using ServiceControlInstaller.Engine.Configuration.ServiceControl;
+using ServiceControlInstaller.Engine.Instances;
+using ServiceControlInstaller.Engine.ReportCard;
+using ServiceControlInstaller.Engine.Validation;
+using UI.AdvancedOptions;
+
+class ForceUpgradeAuditInstanceCommand : AwaitableAbstractCommand<ServiceControlAdvancedViewModel>
+{
+    public ForceUpgradeAuditInstanceCommand(
+        IServiceControlWindowManager windowManager,
+        IEventAggregator eventAggregator,
+        ServiceControlAuditInstanceInstaller serviceControlAuditInstaller) : base(ForcedUpgradeAllowed)
+    {
+        this.windowManager = windowManager;
+        this.eventAggregator = eventAggregator;
+        this.serviceControlAuditInstaller = serviceControlAuditInstaller;
+    }
+
+    [FeatureToggle(Feature.LicenseChecks)]
+    public bool LicenseChecks { get; set; }
+
+    static bool ForcedUpgradeAllowed(ServiceControlAdvancedViewModel model)
+    {
+        var instance = InstanceFinder.ServiceControlAuditInstances().FirstOrDefault(i => i.Name == model.Name);
+
+        //HINT: Force upgrade is available only primary v4 instance, running on RavenDB 3.5
+        return instance != null && instance.Version.Major == 4 && instance.PersistenceManifest.Name != StorageEngineNames.RavenDB;
+    }
+    public override async Task ExecuteAsync(ServiceControlAdvancedViewModel model)
+    {
+        if (await windowManager.ShowMessage("Forced migration",
+                "Do you want to proceed with forced migration to version 5?", "Yes") == false)
+        {
+            return;
+        }
+
+        if (LicenseChecks)
+        {
+            var licenseCheckResult = serviceControlAuditInstaller.CheckLicenseIsValid();
+            if (!licenseCheckResult.Valid)
+            {
+                await windowManager.ShowMessage("LICENSE ERROR", $"Upgrade could not continue due to an issue with the current license. {licenseCheckResult.Message}.  Contact contact@particular.net", hideCancel: true);
+                return;
+            }
+        }
```

Comment:
Remove after rebasing on https://github.com/Particular/ServiceControl/pull/3808 and insert the check using CommandChecks instead.